README.md

```diff
@@ -89,61 +89,143 @@ builds like `gcc` generate a *lot* of output), and builds timing out
 
-# Packages from ~/.travis.yml; realpath is in coreutils in Ubuntu 18.04
-# Plus libtool and pkg-config, which are needed for openocd
-#
-# sudo apt-get install realpath
-sudo apt-get install coreutils
+# Initialize Conda in the shell
+eval "$("$CONDA_PATH/bin/conda" "shell.bash" 'hook' 2> /dev/null)"
 
+# Install `conda-build-prepare`
+python3 -m pip install git+https://github.com/litex-hub/conda-build-prepare@v0.1#egg=conda-build-prepare
+
+# Clone the litex-conda-compilers repository
 git clone https://github.com/litex-hub/litex-conda-compilers.git
 cd litex-conda-compilers
-./.travis/conda-get.sh
-
-# Adapted from .travis/common.sh
-get_built_package() {
-   ./.travis/conda-env.sh render --output "$@" 2>/dev/null | grep conda-bld | grep tar.bz2 | tail -n 1 | sed -e's/-[0-9]\+\.tar/*.tar/' -e's/-git//'
-}
-
-# Anchor the build date/time, so we have predictable versions and filenames
-# amongst related packages, and to make it easy to do package installs.
-#
-# Either to current date/time at the start of the build:
-#
-# export DATE_NUM="$(date -u +%Y%m%d%H%M%S)"
-# export DATE_STR="$(date -u +%Y%m%d_%H%M%S)"
-#
-# Or lock to date/time of the last commit on git, as Travis CI config does
-# (see .travis/common.sh)
-#
-export DATE_TS="$(git log --format=%ct -n1)"
-export DATE_NUM="$(date --date=@${DATE_TS} -u +%Y%m%d%H%M%S)"
-export DATE_STR="$(date --date=@${DATE_TS} -u +%Y%m%d_%H%M%S)"
-
-# Combinations taken from .travis.yml
-TOOLCHAIN_ARCH=lm32
-export PACKAGE TOOLCHAIN_ARCH
-
-cd conda-hdmi2usb-packages
-
-for PACKAGE in binutils gcc/nostdc gcc/newlib; do
-  ./.travis/conda-env.sh build --check "${PACKAGE}"   # Downloads and caches stuff
-  ./.travis/conda-env.sh build         "${PACKAGE}"   # Actually build package
-  CONDA_OUT="$(get_built_package ${PACKAGE})" # Calculate output package
-  ./.travis/conda-env.sh install       "${CONDA_OUT}"
-done
 ```
 
+### Required environment variables
+
+The recipe might require exporting some additional environment variables.
+Such variables must be set before preparing the recipe, which is done
+with one of the commands from the next subsection.
+
+Currently required additional environment variables are:
+* `TOOLCHAIN_ARCH` (by: `binutils`, `gcc/*`, `gdb` and `toolchain/linux-musl`
+recipes) – must contain a supported architecture name, ie. one of these:
+  * `lm32` (excluding `gcc/linux-musl` and `toolchain/linux-musl` recipes),
+  * `or1k`,
+  * `ppc64le` (only `binutils` and `gcc/linux-musl` recipes),
+  * `riscv32`,
+  * `riscv64`,
+  * `sh`.
+
+The optional `DATE_NUM` and `DATE_STR` environment variables are used by the
+recipes to set the `build/number` and `build/string` keys during preparation.
+In case they're unset, recipe preparation provides them with default values
+based on the latest commit's committer date from the repository holding the
+recipe.
+
+### Preparing and building the package
+
+After getting all prerequisites and setting the required variables, it
```

Comment:
I feel there might be misunderstandings about what "local" means. There are three possible setups:

- The "native" Ubuntu environments.
- A container.
- Executing some commands natively and others in a container.

Those three setups are possible either locally or in CI. Hence, 6 total. At first, I thought that you were only considering "native Ubuntu environments" in CI and "a container" locally. Then, I understood that repos might be cloned natively, and other commands executed in a container. Now, I am unsure.

My recommendation would be to create/maintain an specific container image with all the prerequisites. Then, use that image both in CI and locally. That will reduce startup time, and it will provide exactly the same result for all developers/collaborators and CI services.

It might make sense to create litex-hub/containers or symbiflow/containers for maintaining build images. We own the symbiflow namespace in dockerhub (I can transfer/give access to any of you). See ghdl/docker (and ghdl/ghdl) and hdl/containers for practical examples.

Reply:
All of the commands provided have to be executed in the same OS and are the same regardless whether used in a host OS or a container. The commands have been tested mostly on Debian/Ubuntu, but apart from the package installation tool there shouldn't be many problems to use it on many other GNU/Linux OSs.

Did command separation into two blocks made you think that some commands should be executed in a host OS and the others in a container? In the same sentence where you suggested a change there is: 
> The commands from the following subsections are enough to build a package in a clean docker container based on the *ubuntu* Docker image.

but maybe I should stress that the commands should also be executed there? Maybe you want to suggest some change to make everything clear?

Your recommendation seems to be very reasonable but this would be quite a change. I don't think it should be handled by this PR. Maybe you could create an issue pointing out what are the problems with the current situation and your suggestion? Simply to not forget about it as I think it is a good idea.